src/modules/base.ts

```diff
@@ -2,8 +2,14 @@ import HttpClient from "core/http-client";
 
 export default class Base {
   protected readonly http: HttpClient;
+  protected readonly baseUrl: string;
 
-  constructor(httpClient: HttpClient) {
+  constructor(httpClient: HttpClient, baseUrl: string) {
     this.http = httpClient;
+    this.baseUrl = baseUrl;
+
+    if (!this.baseUrl) {
+      throw new Error('baseUrl has to be defined.');
+    }
   }
 }
```

Comment:
@SparoHawk files should end with a extra line at the end?

Reply:
When we setting up the project the configuration for new line at end of file came turned off. We still haven't gone through configuring the project styling entirely.